staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting/validation.go

```diff
@@ -86,7 +86,11 @@ func validate(pth *field.Path, s *structuralschema.Structural, rootSchema *struc
 			} else if errs := apiservervalidation.ValidateCustomResource(pth.Child("default"), s.Default.Object, validator); len(errs) > 0 {
 				allErrs = append(allErrs, errs...)
 			} else if celValidator := cel.NewValidator(s); celValidator != nil {
-				allErrs = append(allErrs, celValidator.Validate(pth.Child("default"), s, s.Default.Object)...)
+				err, remainingBudget := celValidator.Validate(pth.Child("default"), s, s.Default.Object, cel.RuntimeCELCostBudget)
+				allErrs = append(allErrs, err...)
+				if remainingBudget < 0 {
+					return allErrs, nil
+				}
```

Comment:
I expected us to propagate `remainingBudget`, not short-circuit. same comment applies below.